docs/react.md

```diff
@@ -63,7 +73,36 @@ const { channel } = useChannel("your-channel-name", (message) => {
 
 Every time a message is sent to `your-channel-name` it'll be logged to the console. You can do whatever you need to with those messages.
 
-Our react hooks are designed to run on the client-side, so if you are using server-side rendering, make sure that your components which use Ably react hooks are only rendered on the client side.
+> [!NOTE]
+> Our react hooks are designed to run on the client-side, so if you are using server-side rendering, make sure that your components which use Ably react hooks are only rendered on the client side.
+
+### Channel Options
+
+We support providing [ChannelOptions](https://ably.com/docs/api/realtime-sdk/types#channel-options) for the `ChannelProvider` component:
+
+This means you can use features like `rewind`:
+
+```jsx
+<ChannelProvider channelName="your-channel-name" options={{ params: { rewind: '1' } }}>
+  <Component />
+</ChannelProvider>
+```
+
+[Subscription filters](https://ably.com/docs/channels#filter-subscribe) are also supported:
+
+```jsx
+const deriveOptions = { filter: 'headers.email == `"rob.pike@domain.com"` || headers.company == `"domain"`' }
+
+return (
+  <ChannelProvider channelName="your-derived-channel-name" options={{ ... }} deriveOptions={deriveOptions}>
+    <Component />
+  </ChannelProvider>
+)
+```
+
+> [!NOTE]
+> Please note that attempts to publish to a derived channel (the one created or retrieved with a filter expression) will fail. In order to send messages to the channel called _"your-derived-channel-name"_ from the example above, you will need to create another channel instance without a filter expression.
```

Comment:
I think this step for a user would also mean using a separate `AblyProvider`, because derived channel and regular one will have the same name, so they can't be placed inside one `AblyProvider`.
Given that this is a bit advanced and non-obvious, maybe we should add an example for this too?

Reply:
Oh sorry I didn't pick up on this in my initial review. Out of interest, what's the reason for not being able to publish to a derived channel? That seems like quite a disappointing limitation to me 🙁

If there really is no way we can fix that then I think recommending rest publishing is probably better than recommending client side filtering since the whole reason people use filtered channels is so that they don't incur the cost of receiving every message on the channel.

Reply:
>Out of interest, what's the reason for not being able to publish to a derived channel?

@owenpearson I think that's a question for rt team :) derived channel has only subscribe capability right now

Reply:
tbh this should work fine? haven't tested though
```ts
const { ably } = useChannel(...opts);
ably.channels.get(channelNameWithoutDeriveExpression).publish();
```

Reply:
cool I haven't known this, could you look at this PR https://github.com/ably/ably-js/pull/1658/files ? WDYT? I think it will be better if we handle this situation ourselves

Reply:
that looks like an ideal solution :)

Reply:
I assume we decided to go with transient realtime publishing instead of rest to keep what's user is exposed to to one API (Realtime in this case), so user doesn't have to figure out how Rest works, what's the difference and how to set it up in the application?